linode/lke/cluster_test.go

```diff
@@ -58,69 +62,93 @@ func TestReconcileLKENodePoolSpecs(t *testing.T) {
 		},
 		{
 			name: "reuse cluster for resize",
-			provisionedPools: []linodego.LKENodePool{
+			oldSpecs: []lke.NodePoolSpec{
 				{ID: 123, Type: "g6-standard-1", Count: 1},
 				{ID: 124, Type: "g6-standard-1", Count: 10},
 			},
-			specs: []lke.NodePoolSpec{
-				{Type: "g6-standard-1", Count: 9},  // bumped from 1 to 9
-				{Type: "g6-standard-2", Count: 10}, // type changed
+			newSpecs: []lke.NodePoolSpec{
+				{ID: 123, Type: "g6-standard-1", Count: 9},  // bumped from 1 to 9
+				{ID: 124, Type: "g6-standard-2", Count: 10}, // type changed
 			},
-			expectedToDelete: []int{123},
+			expectedToDelete: []int{124},
 			expectedToUpdate: map[int]linodego.LKENodePoolUpdateOptions{
-				124: {Count: 9},
+				123: {Count: 9},
 			},
 			expectedToCreate: []linodego.LKENodePoolCreateOptions{
 				{Type: "g6-standard-2", Count: 10},
 			},
 		},
 		{
 			name: "competing resizes",
-			provisionedPools: []linodego.LKENodePool{
+			oldSpecs: []lke.NodePoolSpec{
 				{ID: 123, Type: "g6-standard-3", Count: 3},
 				{ID: 124, Type: "g6-standard-3", Count: 7},
 				{ID: 126, Type: "g6-standard-3", Count: 4},
 				{ID: 127, Type: "g6-standard-3", Count: 2},
 			},
-			specs: []lke.NodePoolSpec{
-				{Type: "g6-standard-3", Count: 2},
-				{Type: "g6-standard-3", Count: 9},
-				{Type: "g6-standard-3", Count: 8},
-				{Type: "g6-standard-3", Count: 2},
+			newSpecs: []lke.NodePoolSpec{
+				{ID: 123, Type: "g6-standard-3", Count: 2},
+				{ID: 124, Type: "g6-standard-3", Count: 9},
+				{ID: 126, Type: "g6-standard-3", Count: 8},
+				{ID: 127, Type: "g6-standard-3", Count: 2},
 			},
 			expectedToUpdate: map[int]linodego.LKENodePoolUpdateOptions{
-				123: {Count: 2}, // -1
-				124: {Count: 8}, // +1
-				126: {Count: 9}, // +5
+				123: {Count: 2},
+				124: {Count: 9},
+				126: {Count: 8},
```

Comment:
The expectation for node pool updates has changed so the counts needed to be swapped